node/src/chain_spec/finney.rs

```diff
@@ -181,7 +181,27 @@ pub fn finney_mainnet_config() -> Result<ChainSpec, String> {
         balances_issuance,
     ))
     .with_properties(properties)
-    .build())
+    .build();
+
+    // Load and set the code substitute to avoid archive node sync panic
+    // See <https://github.com/opentensor/subtensor/pull/1051>
+    //
+    // Need to do it in this hacky way because the ChainSpec builder doesn't support setting it
+    let manifest_dir = env!("CARGO_MANIFEST_DIR");
+    let blob_path =
+        std::path::Path::new(manifest_dir).join("src/chain_spec/code_substitute_2585476.txt");
```

Comment:
This requires having `code_substitute_2585476.txt` at the specified path at the runtime. You need to embed it into binary using `include_str!`or `include_bytes!` macro.

Reply:
Also, I believe in case of `include_*` macros you can use a relative path.

Reply:
I'm not sure what you mean, `code_substitute_2585476.txt` is checked into Git at the specified path so it should always resolve correctly.

Why are you saying it needs to be embedded into the binary?

Reply:
Because the node is a binary, you can install it in a different location with different directory structure. If you build your node then move the binary from the build directory to another path and run it with `--chain finney` it will panic, because it won't find `code_substitute_2585476.txt`. Moreover, `CARGO_MANIFEST_DIR` won't be available at that time. For the node to work correctly, you need either provide this file and `CARGO_MANIFEST_DIR`, change the path to relative to binary and also provide the file or simply embed it into the binary.

Reply:
Just imagine you installed the node at `/foo/bar/node-subtensor` and what will happen then if you run it like `/foo/bar/node-subtensor --chain finney`. Not taking `CARGO_MANIFEST_DIR` availability into account, even if there were a relative to binary path, the binary would expect `code_substitute_2585476.txt` at `/foo/bar/src/chain_spec/code_substitute_2585476.txt`.

Reply:
I do see your point there is a benefit, but the chain spec is almost 1mb, it would bloat the binary a lot if we added it. 

I can't think of when we would want to generate the chain spec outside of the repo using the bare binary, so I think the costs outweigh the benefits to embed it. Let me know if you think I'm missing something though.

Reply:
Yeah, unfortunately. I mentioned this in the issue, where I suggested embedding the genesis blob into the binary. For validator's and standard nodes it's fine, given the size of the blockchain and that you have the chainspec file behind the binary anyway. But for the lightnodes (if we will have one), it's definitely not good. It's a subject for a discussion though, as we have to implement it separately from a normal node using something like https://github.com/smol-dot/smoldot, don't we?
 
If we go the way when we should solve the path at the runtime, then instead of panicing, it would be better to extend the `build-spec` CLI command with a required argument that accepted the genesis file. At least that would persist the portability. Otherwise, using the script as you initially did is a better solution, than relying on a file that may not exist at the runtime.

Reply:
> But for the lightnodes (if we will have one), it's definitely not good

we don't support lightnodes

> If we go the way when we should solve the path at the runtime, then instead of panicing, it would be better to extend the build-spec CLI command with a required argument that accepted the genesis file.

it might be a bit difficult to extend the build-spec cli. I checked the release binary size and it's already 82mb, so I think adding another 1mb is fine. I will just do that for now  :)

Reply:
Hey @ales-otf, please take another look